README.md

```diff
@@ -1,353 +1,228 @@
-```
-sudo apt install autoconf bison build-essential clang clang-tidy flex \
-    g++ gcc iverilog liblpsolve55-dev libssl-dev libtool libxerces-c3.2 \
-    libxerces-c-dev lld make ninja-build python zlib1g zlib1g-dev
+* `ninja-build` (preferred) or `make`
+
+The following command can be used to install all of these dependencies regardless of what exactly will be used to compile Utopia HLS:
+```bash
+sudo apt install cmake build-essential clang g++ gcc liblpsolve55-dev lld make ninja-build
 ```
 
-### CMake Installation
+### Precompiled CIRCT & MLIR Installation
 
-```
-cd <workdir>
-wget https://cmake.org/files/v3.27/cmake-3.27.3-linux-x86_64.tar.gz
-tar xzf cmake-3.27.3-linux-x86_64.tar.gz
-rm -rf cmake-3.27.3-Linux-x86_64.tar.gz
-cd cmake-3.27.3-Linux-x86_64
-./bootstrap
-make
-sudo make install
-```
+Utopia HLS utilizes SystemVerilog-generation capabilities from CIRCT project, which itself is based on LLVM Multi-Level Intermediate Representation (MLIR) project.
 
-### Z3 Installation
+**[!!!]Currently supported CIRCT release: 1.72.0, Apr 5 2024 ([link](https://github.com/llvm/circt/releases/tag/firtool-1.72.0))[!!!]**
 
-```
-cd <workdir>
-git clone https://github.com/Z3Prover/z3.git
-cd z3
-git checkout 013d5dc
-git submodule update
-python scripts/mk_make.py
-cd build
-make
-sudo make install
-```
-If you would like to install Z3 to a non-standard location,
-please set `Z3_DIR` environment variable to Z3 build/installation directory.
+For both CIRCT and MLIR projects it is possible to download precompiled libraries (this approach is more convenient and saves time), but both projects can also be compiled from sources.
 
-### CIRCT Installation
+Precompiled releases of CIRCT include the precompiled LLVM MLIR releases as well: visit the corresponding [page](https://github.com/llvm/circt/releases) to see all available releases. 
 
-LLVM requires a significant amount of RAM (about 8 Gb or more) to build.
-Please take this into account while moving through the guide.
+**Note that <ins>it's the libraries that are required</ins>, not the binary executables.**<br>
+Look for the archives which have the following names:
+`circt-full-static-<ARCH>` or `circt-full-shared-<ARCH>` for static and dynamic libraries respectively.
 
-#### Check out LLVM and CIRCT repos
+**[!!!]Current releases have an inconsistency in their configuration files[!!!]:**<br>
+after downloading the chosen release archive, extract the files inside and look for the file `lib/cmake/mlir/MLIRTargets.mlir`.
 
-```
-cd <workdir>
-git clone --recursive https://github.com/circt/circt.git
-cd circt
-git checkout 5c31646
+Open `MLIRTargets.mlir` with a text editor of your choice and look the `_NOT_FOUND_MESSAGE_targets` near the end of the file:<br>
+for 1.72.0, it's line **3012** for `circt-full-static-linux-x64`-archive and line **3008** for `circt-full-shared-linux-x64.tar`-archive.
+
+Remove all the `"CIRCT*"`-entries from the corresponding `foreach`-statement and save the file.
```

Comment:
Fixed. I have already thought of using some hand-made Bash-scripts with `sed`, but given how many times `CIRCT*` substrings are present in the configuration file, it might be hard to write a correct script.